cli/src/commands/import/csv.ts

```diff
@@ -69,100 +67,202 @@ export default class ImportCSV extends BaseCommand<typeof ImportCSV> {
+          throw new Error('Failed to parse CSV file');
+        }
+        const dbBranchName = `${database}:${branch}`;
+        const importResult = await xata.import.importBatch(
+          // @ts-ignore
+          { dbBranchName: dbBranchName, region, workspace: workspace, database },
+          { columns: parseResults.columns, table, batch: parseResults }
+        );
+        importSuccessCount += importResult.successful.results.length;
+        if (importResult.errors) {
+          const formattedErrors = importResult.errors.map(
+            (error) => `${error.error}. Record: ${JSON.stringify(error.row)}`
+          );
+          const errorsToLog = formattedErrors.slice(0, Math.abs(ERROR_CONSOLE_LOG_LIMIT - errors.length));
+          for (const error of errorsToLog) {
+            this.logToStderr(`Import Error: ${error}`);
+          }
+          errors.push(...formattedErrors);
+        }
+        progress = Math.max(progress, meta.estimatedProgress);
+        this.info(
+          `${importSuccessCount} rows successfully imported ${errors.length} errors. ${Math.ceil(
+            progress * 100
+          )}% complete`
         );
       }
     });
-    if (error) {
-      return process.exit(1);
+    if (errors.length > 0) {
+      await writeFile(ERROR_LOG_FILE, errors.join('\n'), 'utf8');
+      this.log(`Import errors written to ${ERROR_LOG_FILE}`);
+    }
+    fileStream.close();
+    this.success('Completed');
+    process.exit(0);
+  }
+
+  databaseInfo: Awaited<ReturnType<typeof this.getParsedDatabaseURLWithBranch>> | null = null;
+
+  async parseDatabase() {
+    if (this.databaseInfo) {
+      return this.databaseInfo;
     }
+    const { flags } = await this.parseCommand();
+    const databaseInfo = await this.getParsedDatabaseURLWithBranch(flags.db, flags.branch, true);
+    this.databaseInfo = databaseInfo;
+    return databaseInfo;
+  }
 
-    if (compare.missingTable) {
-      if (!create) {
-        const response = await this.prompt({
-          type: 'confirm',
-          name: 'confirm',
-          message: `Table ${table} does not exist. Do you want to create it?`,
-          initial: false
-        });
-        if (!response.confirm) return false;
-      }
-    } else if (compare.missingColumns.length > 0) {
-      if (!create) {
-        const response = await this.prompt({
+  async migrateSchema({
+    table,
+    columns,
+    create
+  }: {
+    table: string;
+    columns: Column[];
+    create: boolean;
+  }): Promise<void> {
+    const xata = await this.getXataClient();
+    const { workspace, region, database, branch } = await this.parseDatabase();
+    const { schema: existingSchema } = await xata.api.branches.getBranchDetails({
+      workspace,
+      region,
+      database,
+      branch
+    });
+    const newSchema = {
+      tables: [
+        ...existingSchema.tables.filter((t) => t.name !== table),
+        { name: table, columns: columns.filter((c) => c.name !== 'id') }
+      ]
+    };
```

Comment:
Something to keep in mind for the UI and maybe a warning in the CLI would be appreciated.

If you had a column with type `int` and now you attempt to upload a file and it that has been guessed as `float`, it will delete the existing column with any values and create a new one.